### ISSUE TITLE
Fn::If gives false alarm on Array value of S3 Bucket CorsRule

### ISSUE THREAD

**Issue by exoego** [martysweet/cfn-lint#233]
[`AllowdOrigins` properties for Amazon S3 Bucket CorsRule accepts `Array` value](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-cors-corsrule.html#cfn-s3-bucket-cors-corsrule-allowedorigins).
My CF tempalte uses `Fn::If` to switch `AllowedOrigins` depending on environment like below:

```json
    "MyBucket": {
      "Type": "AWS::S3::Bucket",
      "Properties": {
        "CorsConfiguration": {
          "CorsRules": [
            {
              "AllowedOrigins": {
                "Fn::If": [
                  "IsDev",
                  [
                    "https://dev1.example.com",
                  ],
                  [
                    "https://foo.prod1.example.com",
                    "https://bar.prod2.example.com",
                  ]
                ]
              },
```

This template is valid and works  fine in CloudFormation.

However, `cfn-lint` gives critical error on this like 
```
1 crit
Resource: Resources > S3Annotation > Properties > CorsConfiguration > CorsRules > 0 > AllowedOrigins
Message: Fn::If does not allow 0 as a nested function within an array
Documentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-conditions.html#intrinsic-function-reference-conditions-if
```

I think this is false alarm. 
Suspect code is
https://github.com/martysweet/cfn-lint/blob/7975480ab9d77bef377b47aef805a584a5ebd2d5/src/validator.ts#L1528-L1537
